examples/darwin-framework-tool/commands/common/CertificateIssuer.mm

```diff
@@ -0,0 +1,157 @@
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#import "CertificateIssuer.h"
+#import "CHIPToolKeypair.h"
+
+#include <lib/support/logging/CHIPLogging.h>
+
+@interface CertificateIssuer ()
+- (MTRCertificateDERBytes _Nullable)issueOperationalCertificateForNodeID:(NSNumber *)nodeID
+                                                                fabricID:(NSNumber *)fabricID
+                                                               publicKey:(SecKeyRef)publicKey
+                                                                   error:(NSError * _Nullable)error;
+@end
+
+@implementation CertificateIssuer
+
++ (CertificateIssuer *)sharedInstance
+{
+    static CertificateIssuer * certificateIssuer = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        certificateIssuer = [[CertificateIssuer alloc] init];
+    });
+    return certificateIssuer;
+}
+
+- (instancetype)init
+{
+    if (!(self = [super init])) {
+        return nil;
+    }
+
+    _rootCertificate = nil;
+    _ipk = nil;
+    _signingKey = nil;
+    _fabricID = nil;
+    _nextNodeID = nil;
+    _shouldSkipAttestationCertificateValidation = NO;
+
+    return self;
+}
+
+- (void)startWithStorage:(id<MTRStorage>)storage
+                   error:(NSError * _Nullable)error
+{
+    __auto_type * signingKey = [[CHIPToolKeypair alloc] init];
+
+    auto err = [signingKey createOrLoadKeys:storage];
+    if (CHIP_NO_ERROR != err) {
+        error = [NSError errorWithDomain:@"Error" code:0 userInfo:@{ @"reason" : @"Error creating or loading keys" }];
+        return;
+    }
+
+    __auto_type * rootCertificate = [MTRCertificates createRootCertificate:signingKey issuerID:nil fabricID:nil error:&error];
+    if (nil == rootCertificate) {
+        error = [NSError errorWithDomain:@"Error" code:0 userInfo:@{ @"reason" : @"Error creating root certificate" }];
+        return;
+    }
+
+    _rootCertificate = rootCertificate;
+    _signingKey = signingKey;
+    _ipk = [signingKey getIPK];
+}
+
+- (MTRCertificateDERBytes _Nullable)issueOperationalCertificateForNodeID:(NSNumber *)nodeID
+                                                                fabricID:(NSNumber *)fabricID
+                                                                   error:(NSError * _Nullable)error
+{
+    __auto_type * publicKey = _signingKey.publicKey;
+    return [self issueOperationalCertificateForNodeID:nodeID fabricID:fabricID publicKey:publicKey error:error];
```

Comment:
This is using the private key of our signing cert as the private key of our NOC?  That seems a little odd...